cli/command/stack/loader/loader.go

```diff
@@ -104,6 +105,19 @@ func GetConfigDetails(composefiles []string, stdin io.Reader) (composetypes.Conf
 func buildEnvironment(env []string) (map[string]string, error) {
 	result := make(map[string]string, len(env))
 	for _, s := range env {
+		if runtime.GOOS == "windows" && len(s) > 0 {
+			// cmd.exe can have special environment variables which names start with "=".
+			// They are only there for MS-DOS compatibility and we should ignore them.
+			// See TestBuildEnvironment for examples.
+			//
+			// https://ss64.com/nt/syntax-variables.html
+			// https://devblogs.microsoft.com/oldnewthing/20100506-00/?p=14133
+			// https://github.com/docker/cli/issues/4078
+			if s[0] == '=' {
```

Comment:
I didn't make this one an explicit blacklist of variables, because:
1. MS documentation explicitly states that:
> The name of an environment variable cannot include an equal sign (=).
https://learn.microsoft.com/en-us/windows/win32/procthread/environment-variables

so user can't deliberately set such variable nor anyone other than `cmd.exe` should actually set such variables.

2. I don't think we want to implement the parser for the drive-specific CWD format (or hardcode every possible drive letter).
3. We don't want to play whack-a-mole if there are some other undocumented variables that can start with =.

Reply:
> 1. MS documentation explicitly states that:
> 
> > The name of an environment variable cannot include an equal sign (=).
> > https://learn.microsoft.com/en-us/windows/win32/procthread/environment-variables

Interesting, thanks!

So;

- Perhaps we should include that link (and description that a user cannot set it) in the comment
- From the above, I wonder indeed if this is something that `os.Environ()` should take into account (if that would make sense) and possibly filter out those "env-vars" (might be worth opening a ticket for in Golang, and see if there's opinions on that)

Reply:
There look to be existing Golang issues related to this:
* https://github.com/golang/go/issues/49886
* https://github.com/golang/go/issues/49882